### ISSUE TITLE
[feature-request] encryption

### ISSUE THREAD

**Issue by revuwa** [mpcjanssen/simpletask-android#303]
Thanks a lot for your enthusiasm to make and provide this awesome app, which I really like to use as often as I can.

Because simpletask uses a todo.txt (and optionally a done.txt) to store the tasks, I thought this could be a great basis to implement a basic encryption feature (to protect your sensitive data and your privacy), because I've seen such feature for simple text files in another app, called 'miniNoteViewer' (https://play.google.com/store/apps/details?id=jp.gr.java_conf.hatalab.mnv), source code: https://bitbucket.org/dalb8/mininoteviewer/src (license: GPLv2+).

Another example, which is using password-based encryption (PKCS#5) with AES-256, (instead of the blowfish algorithm as miniNoteViewer do), would be the 'CrocodileNote' app (https://play.google.com/store/apps/details?id=groomiac.crocodilenote), source code: https://github.com/Groomiac/crocodilenote/ (license: GPLv3).

Probably some code snippets could be taken as a kind of a template for simpletask?
I've created a little slide show to indicate what I'm visually thinking of:

Settings:
![01_setting](https://cloud.githubusercontent.com/assets/7300276/10088368/b2f5616e-631a-11e5-9395-d98079a4a5c0.png)

on activation:
![02_set_pw](https://cloud.githubusercontent.com/assets/7300276/10088378/be7e1c38-631a-11e5-859b-2831861b64e3.png)

on opening simpletask with encrypted todo.txt (and/or done.txt):
![03_input_pw](https://cloud.githubusercontent.com/assets/7300276/10088417/f63a577c-631a-11e5-9b8f-e7ff1d6006fa.png)

if you type in the wrong password:
![04_pw_wrong](https://cloud.githubusercontent.com/assets/7300276/10088430/07ab0a1a-631b-11e5-9773-b5abdb957376.png)

Such a feature I'm really missing and would be very cool, if it would come in any form in the future.

**Comment by mpcjanssen** [comment]
Thanks for the nice mockups. But I don't think this is going to be added any time soon.

**Comment by smichel17** [comment]
I'm confused about the use case for encryption. Here's what I could think of...

1) If someone steals my device, I don't want them to be able to read my to-do's. I think android's FDE would work better here, so someone who steals your device can't read *anything*. Granted, good FDE (different lock screen and disk passwords) currently requires a rooted phone.

2) I don't want google / dropbox to be able to read my to-do's. I think software like boxcryptor (just the first thing I found, didn't check if there's floss alternatives) would do the job better here, encrypting *anything* you want to store in dropbox as it's uploaded, then decrypting it locally.

Yes, there might be some issues with both of those right now, but in the long term they're both more robust solutions and more modular -> easier to maintain. If @revuwa and @mpcjanssen agree, close this issue?

**Comment by mpcjanssen** [comment]
@smichel17 Agreed however integrating a encrypting tool with another android application is not possible right now. It might be possible to do this for Cloudless with FDE but on Dropbox it's impossible. 

Therefore I do not want to close this issue just yet.

**Comment by revuwa** [comment]
@smichel17 
To your first point. You're right, (and I'm using FDE), but this doesn't protect you against losing the todo.txt to other people/services/companies through security issues on your device (or some nosy apps) or just your backup system, because it's a simple text-file, and you could use simpletask for much more than just regular todo stuff...

This is probably the same point, you meant in your second one ("[...] don't want google / dropbox to be able to read my to-do's [...]")

I don't know, I just thought the 'miniNoteViewer' app simply encrypts the plaintext-text-file, so I hoped this (open source) code could help/lead to bring this functionality to this great app here, because I still believe this would be a great privacy/security feature, no other (actively developed FOSS) app could offer. In other words, I still would love to see such a functionality.

**Comment by mpcjanssen** [comment]
@revuwa just encrypting the file is not enough. You will also need to encrypt the internal databases to be really save. This makes it a bit more difficult.

I will check the miniNoteViewer code for inspiration.

**Comment by vojtechkral** [comment]
@revuwa I doubt there's much point in trying to protect the todo.txt file against the users phone's ecosystem. If the user installs an app that reads their files and sends the data somewhere, their privacy has already been compromised and IMHO there is little point trying to fight this in Simpletask. This really seems beyond Simpletask and up to the phone's system to encrypt data and/or protect against malicious apps.

The only use case I'd consider valid would be encrypting the todo.txt before uploading it to Dropbox.

Just my two cents.

**Comment by smichel17** [comment]
I'd second that. It should be android's responsibility to provide a sandboxed space for apps.

There is an argument "Android isn't doing it so we need to fill in the gaps in the meantime." If the focus of this app were privacy (Signal does it), or if we had a bigger dev team, I'd agree with that argument, but right now @mpcjanssen is doing >95% of commits.

Therefore, one valid use case.
